app/forms/flood_risk_engine/steps/base_address_form.rb

```diff
@@ -49,11 +49,16 @@ def assign_via_uprn
         if address_attributes
           validator = AddressValidator.new(address_attributes)
 
-          @address_to_assign = Address.new(address_attributes) if validator.validate
-
-          logger.debug("AddressValidator returned [#{validator.valid?}]")
-
-          logger.debug("LocalAuthority address now #{@address_to_assign.inspect}")
+          if validator.validate
+            @address_to_assign = Address.new(address_attributes)
+            logger.debug("LocalAuthority address now #{@address_to_assign.inspect}")
+          else
+            # not too sure how to merge this validation into Form automatically so for now
+            # manually clone error messages
+            logger.debug("Address Valid #{validator.valid?} Errors [#{validator.errors.inspect}]")
+
+            validator.errors.each { |_e, m| errors.add :address, m }
```

Comment:
If errors returns a hash, iterating over the values would remove the need to have the unused `_e` variable:

`validator.errors.values.each { |message| errors.add :address, message }`